client/cmd/dexc-desktop/pkg/pkg-darwin.sh

```diff
@@ -0,0 +1,334 @@
+
+# App information
+APP_NAME="Decred DEX Client"
+VOLUME_NAME="Decred DEX Client ${VER}${META:++${META}}"
+
+# Filepaths to important directories.
+SRC_DIR="../src"
+SCRIPTPATH="$( cd -- "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+INSTALLERS_DIR="$SCRIPTPATH/installers"
+APP_DIR="${SCRIPTPATH}/${APP_NAME}.app"
+CONTENTS_DIR="${APP_DIR}/Contents"
+RESOURCES_DIR="${CONTENTS_DIR}/Resources"
+APP_EXCE_DIR="${CONTENTS_DIR}/MacOS"
+
+# For fancy disk image container icon and background.
+ICON_FILE_NAME=dexc-icon.icns
+ICON_FILE="${SRC_DIR}/dexc-icon.icns"
+VOLUME_ICON_FILE="${ICON_FILE}" # use the same icon for the volume.
+BACKGROUND_FILE_NAME="dexc-installer-bg.png"
+BACKGROUND_FILE="${SRC_DIR}/dexc-installer-bg.png"
+APPLESCRIPT_FILE="build-template.applescript" # will be created during build process.
+
+# For dmg container and folder icon positioning .
+WINX=80
+WINY=100
+WINW=850
+WINH=400
+ICON_SIZE=80
+TEXT_SIZE=16
+
+# For .app folder positioning.
+APP_FOLDER_X=230
+APP_FOLDER_Y=305
+
+# For the Applications folder positioning.
+APPLICATIONS_FOLDER_X=590
+APPLICATIONS_FOLDER_Y=310
+
+# For use as params to hdiutil.
+FORMAT="UDZO"
+FILESYSTEM="HFS+" # or APFS
+FILESYSTEM_ARGUMENTS="-c c=64,a=16,e=16" # will be set to empty if using APFS
+IMAGEKEY="-imagekey zlib-level=9" # This is only used for disk format UDZO, set to empty for others.
+
+# For signing.
+SIGNATURE=""
+NOTARIZE=""
+
+# Helper funtions
+
+function hdiutil_detach() {
+	# Unmount
+	echo "Unmounting disk image..."
+	hdiutil detach "$1"
+}
+
+# Use Megabytes since hdiutil fails with very large byte numbers.
+function blocks_to_megabytes() {
+	# Add 1 extra MB, since there's no decimal retention here
+	MB_SIZE=$((($1 * 512 / 1000 / 1000) + 1))
+	echo $MB_SIZE
+}
+
+function get_size() {
+	# Get block size in disk
+	if [[ $OS_MAJOR_VERSION -ge 12 ]]; then
+		bytes_size=$(du -B 512 -s "$1")
+	else
+		bytes_size=$(du -s "$1")
+	fi
+	bytes_size=$(echo $bytes_size | sed -e 's/	.*//g')
+	echo $(blocks_to_megabytes $bytes_size)
+}
+
+# signs the dmg file given to it as an argument.
+function sign() {
+	if [[ -n "${SIGNATURE}" && "${SIGNATURE}" != "-null-" ]]; then
+		echo "Codesign started"
+		codesign -s "${SIGNATURE}" "$1"
+		dmgsignaturecheck="$(codesign --verify --deep --verbose=2 --strict "$1" 2>&1 >/dev/null)"
+		if [ $? -eq 0 ]; then
+			echo "The disk image is now codesigned"
+		else
+			echo "The signature seems invalid.."
+			exit 1
+		fi
+	fi
+}
+
+# notarizes the dmg file given to it as an argument.
+function notarize() {
+	if [[ -n "${NOTARIZE}" && "${NOTARIZE}" != "-null-" ]]; then
+		echo "Notarization started"
+		xcrun notarytool submit "$1" --keychain-profile "${NOTARIZE}" --wait
+		echo "Stapling the notarization ticket"
+		staple="$(xcrun stapler staple "$1")"
+		if [ $? -eq 0 ]; then
+			echo "The disk image is now notarized"
+		else
+			echo "$staple"
+			echo "The notarization failed with error $?"
+			exit 1
+		fi
+	fi
+}
```

Comment:
These aren't been used atm since a $99 dev account is required to get an Apple dev signature for code signing. Notarizing the dexc-desktop dmg is debatable but IMO we might not need it.